server/src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -135,6 +137,20 @@ public CompressedXContent(byte[] data) throws IOException {
         this(new BytesArray(data));
     }
 
+    public static CompressedXContent fromJSON(String json) throws IOException {
```

Comment:
Added this to normalize the whitespaces away when doing the string to `CompressedXContent` conversion. This saves quite a bit of parsing if requests come from external sources (Beats etc.) which may or may not use formats including some whitespaces.